website/content/docs/concepts/filtering/index.mdx

```diff
@@ -0,0 +1,128 @@
+<Selector> is not empty
+
+// Contains checks or Substring Matching
+"<Value>" in <Selector>
+"<Value>" not in <Selector>
+<Selector> contains "<Value>"
+<Selector> not contains "<Value>"
+
+// Regular Expression Matching
+<Selector> matches "<Value>"
+<Selector> not matches "<Value>"
+```
+
+### Selectors
+
+Selectors are used by matching operators to create an expression. Inputs to
+filter expressions are JSON (or JSON-derived); selectors therefore use [JSON
+Pointer](https://tools.ietf.org/html/rfc6901) to select values from the input.
+Each selector must be enclosed in quotes and contain a valid JSON Pointer path,
+including leading slash (`/`).
+
+```text
+// Selects the value `zipzap` from the input `{ "foo": { "bar": "zipzap" } }`
+"/foo/bar"
+```
+
+### Values
+
+Values are used by matching operators to create an expression. Values can be any
+valid selector, a number, or a string. It is best practice to quote values.
+
+Numbers can be base 10 integers or floating point numbers.
+
+When quoting strings, they may either be enclosed in double quotes or backticks.
+When enclosed in backticks they are treated as raw strings and escape sequences
+such as `\n` will not be expanded.
+
+## Connect expressions
+
+There are several methods for connecting expressions, including
+
+- Logical `or`
+- Logical `and`
+- Logical `not`
+- Grouping with parenthesis
+- Matching expressions
+
+```text
+// Logical Or - evaluates to true if either sub-expression does
+<Expression 1> or <Expression 2>
+
+// Logical And - evaluates to true if both sub-expressions do
+<Expression 1 > and <Expression 2>
+
+// Logical Not - evaluates to true if the sub-expression does not
+not <Expression 1>
+
+// Grouping - Overrides normal precedence rules
+( <Expression 1> )
+
+// Inspects data to check for a match
+<Matching Expression 1>
+```
+
+Standard operator precedence can be expected for the various forms. For example,
+the following two expressions would be equivalent.
+
+```text
+<Expression 1> and not <Expression 2> or <Expression 3>
+
+( <Expression 1> and (not <Expression 2> )) or <Expression 3>
+```
+
+## Performance
+
+Filters are executed on Vault nodes and therefore will consume some amount
+of CPU time on the node.
+
+<Note title="Regular expressions">
+  Using multiple/complex expressions including regular expressions
+  (regex) will have a larger impact on performance than fewer/simpler filters.
+</Note>
+
+We recommend that all filters are tested in pre-production environments in order
+to ensure correctness.
+
+Ideally you should [codify your management of Vault](/vault/tutorials/operations/codify-mgmt-vault-terraform)
+using tools such as [Terraform](https://www.terraform.io/), to prevent accidentally enabling an audit device
+in a production environment with untested/incorrect settings.
+
+Finally, always ensure you profile production-like workloads within your pre-production
+environments in order to accurately assess the performance of Vault.
```

Comment:
Same question here. The last two points seem like general administrative/performance advice, not anything that's specific to filtering. I'm not saying it isn't potentially useful, just that this doesn't seem to be the right place for it.

Reply:
This is because filtering is an advanced topic, and we want Operators to ensure they've profiled workloads on non-prod setups before going to prod. If they've done something like incorrectly configured lots of filters, this could lead to Vault trying to audit a lot and causing performance degradation. So we want to make sure we've given a friendly reminder that when you're in the world of filtering, you should be doing things by the book.